src/clock.rs

```diff
@@ -0,0 +1,649 @@
+• DCOCLK: Internal digitally controlled oscillator (DCO) with programmable frequencies: 3-MHz default
+(Prog. Freq. 1,5  3  6  12  24  48 MHz)
+• VLOCLK: Internal very-low-power low-frequency oscillator (VLO) with 9.4-kHz typical frequency
+• REFOCLK : Internal, low-power low-frequency oscillator (REFO) with selectable 32.768-kHz or 128-kHz
+• MODCLK: Internal low-power oscillator with 25-MHz typical frequency
+• SYSOSC: Internal oscillator with 5-MHz typical frequency
+
+ACLK: Auxiliary clock
+ACLK is software selectable as LFXTCLK, VLOCLK, or REFOCLK.
+ACLK can be divided by 1, 2, 4, 8, 16, 32, 64, or 128.
+ACLK is restricted to maximum frequency of operation of 128 kHz.
+
+MCLK: Master clock
+MCLK is software selectable as LFXTCLK, VLOCLK, REFOCLK, DCOCLK, MODCLK, or HFXTCLK.
+MCLK can be divided by 1, 2, 4, 8, 16, 32, 64, or 128.
+
+HSMCLK: Subsystem master clock.
+HSMCLK is software selectable as LFXTCLK, VLOCLK, REFOCLK, DCOCLK, MODCLK, HFXTCLK.
+HSMCLK can be divided by 1, 2, 4, 8, 16, 32, 64, or 128.
+
+SMCLK: Low-speed subsystem master clock.
+SMCLK uses the HSMCLK clock resource selection for its clock resource.
+SMCLK can be divided independently from HSMCLK by 1, 2, 4, 8, 16, 32, 64, or 128.
+SMCLK is limited in frequency to half of the rated maximum frequency of HSMCLK.
+
+BCLK: Low-speed backup domain clock.
+BCLK is software selectable as LFXTCLK and REFOCLK and is used primarily in the backup domain.
+BCLK is restricted to a maximum frequency of 32.768 kHz
+*/
+
+use cortex_m::interrupt;
+
+use crate::time::Hertz;
+use pac::cs::csctl0::DCORSEL_A;
+use pac::cs::csctl2::HFXTFREQ_A;
+pub use pac::cs::csctl1::{SELS_A, SELA_A, SELM_A, DIVM_A, DIVS_A};
+use pac::cs::csclken::REFOFSEL_A;
+use pac::CS;
+
+pub trait CsExt {
+    fn constrain(self) -> ClockConfig<NoClockDefined, NoClockDefined>;
+}
+
+impl CsExt for CS {
+    fn constrain(self) -> ClockConfig<NoClockDefined, NoClockDefined> {
+        ClockConfig::new(self)
+    }
+}
+
+/// Typestate for `ClockConfig` that represents unconfigured clocks
+pub struct NoClockDefined;
+/// Typestate for `ClockConfig` that represents a configured MCLK
+pub struct MclkDefined(MclkSel);
+/// Typestate for `ClockConfig` that represents a configured SMCLK
+pub struct SmclkDefined(DIVS_A);
+
+/// MODCLK frequency
+pub const MODCLK: u32 = 25_000_000;
+/// VLOCLK frequency
+pub const VLOCLK: u32 = 9_400;
+/// LFXTCLK frequency
+pub const LFXTCLK: u32 = 32_768;
+/// HFXT min frequency
+pub const HFXTMINCLK: u32 = 1_000_000;
+/// HFXT max frequency
+pub const HFXTMAXCLK: u32 = 48_000_000;
+
+///Selectable HFXTCLK frequencies
+#[derive(Debug, Clone, Copy)]
+pub enum HfxtclkFreqSel {
+    // 1-4 MHz
+    _1_4MHz,
+    // 4-8 MHz
+    _4_8MHz,
+    // 8-16 MHz
+    _8_16MHz,
+    // 16-24 MHz
+    _16_24MHz,
+    // 24-32 MHz
+    _24_32MHz,
+    // 32-40 MHz
+    _32_40MHz,
+    // 40-48 MHz
+    _40_48MHz,
```

Comment:
Self-explanatory, no need for comments

```suggestion
    _1_4MHz,
    _4_8MHz,
    _8_16MHz,
    _16_24MHz,
    _24_32MHz,
    _32_40MHz,
    _40_48MHz,
```

Reply:
Ok, It will be removed.